### PR TITLE
Better 'describe' command output, supersedes #134.

### PR DESCRIPTION
**Pull request by seancribbs**
Example:

```
$ bin/riak config describe bitcask.expiry
Documentation for bitcask.expiry
By default, Bitcask keeps all of your data around. If your
data has limited time-value, or if for space reasons you need to
purge data, you can set the `expiry` option. If you needed to
purge data automatically after 1 day, set the value to `1d`.
Default is: `off` which disables automatic expiration

   Datatype     : [{atom,off},{duration,s}]
   Default Value: off
   Set Value    : undefined
   app.config   : bitcask.expiry_secs 
```

Also, sometimes the default value will be wrapped in quotes if it's a string. We need a better name for "app.config" too.

This pull-request's output looks more like this:

```
Documentation for bitcask.expiry
By default, Bitcask keeps all of your data around. If your
data has limited time-value, or if for space reasons you need to
purge data, you can set the `expiry` option. If you needed to
purge data automatically after 1 day, set the value to `1d`.
Default is: `off` which disables automatic expiration

   Valid Values: 
     - the text "off"
     - a time duration with units, e.g. '10s' for 10 seconds

   Default Value: off
   Set Value    : undefined
   Internal key : bitcask.expiry_secs
```

### CODE REVIEW

File: src/cuttlefish_escript.erl [hunk 1]
```diff
@@ -164,13 +165,20 @@ describe(ParsedArgs, Query) when is_list(Query) ->
...
-            ?STDOUT("   Default Value: ~p", [cuttlefish_mapping:default(Match)]),
+            ValidValues = [
+                            ?FORMAT("~n     - ~s", [cuttlefish_conf:pretty_datatype(Type)]) ||
+                              Type <- lists:flatten([cuttlefish_mapping:datatype(Match)]) ],
+            ?STDOUT("   Valid Values: ~s~n", [ValidValues]),
+            ?STDOUT("   Default Value: ~s", [format_datatype(cuttlefish_mapping:default(Match), cuttlefish_mapping:datatype(Match))]),
 
             Conf = load_conf(ParsedArgs),
-            ConfiguredValue = proplists:get_value(QDef, Conf, undefined),
+            ConfiguredValue = case proplists:get_value(QDef, Conf, undefined) of
```
Comment (joedevivo) [inline src/cuttlefish_escript.erl]: What do you think about making a `format_datatype(undefined, _)` clause so we don't need this case statement?

Comment (seancribbs) [inline src/cuttlefish_escript.erl]: Good idea

Comment (seancribbs) [inline src/cuttlefish_escript.erl]: Oh, but what if one of the enum values is 'undefined'. Bleah

Comment (joedevivo) [inline src/cuttlefish_escript.erl]: That should be fine because 'undefined' and undefined from an enum are bolth pretty printed the same way, right?

Comment (seancribbs) [inline src/cuttlefish_escript.erl]: Yeah but I'd rather distinguish between the two. I could switch it to `lists:keyfind`, which returns `false` if the tuple is not present.

Comment (seancribbs) [inline src/cuttlefish_escript.erl]: What if it didn't print anything at all?

File: test/cuttlefish_test_group_leader.erl [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        {io_request, From, ReplyAs, Req} ->
+            P = process_flag(priority, normal),
+            %% run this part under normal priority always
+            NewOutput = io_request(From, ReplyAs, Req, Output),
+            process_flag(priority, P),            
+            group_leader_loop(Runner, NewOutput);        
+        {get_output, Ref, From} ->
+            From ! {Ref, queue:to_list(Output)},
+            group_leader_loop(Runner, Output);
+        stab ->
```
Comment (joedevivo) [inline test/cuttlefish_test_group_leader.erl:43]: lol

Comment (seancribbs) [inline test/cuttlefish_test_group_leader.erl:43]: Trolling yourself in the future

**Comment by joedevivo** [comment]
+1, get it in!

**Pull request merged**